sdk/storage/sample/datalake_getting_started.cpp

```diff
@@ -3,11 +3,80 @@
 
 #include "datalake/datalake.hpp"
 #include "samples_common.hpp"
+#include <iostream>
 
 SAMPLE(DataLakeGettingStarted, DataLakeGettingStarted)
 void DataLakeGettingStarted()
 {
+  // Please note that you can always reference test cases for advanced usages.
+
   using namespace Azure::Storage::Files::DataLake;
-  auto client = ServiceClient::CreateFromConnectionString(GetConnectionString());
-  auto response = client.ListFileSystems();
+
+  std::string fileSystemName = "sample-file-system";
+  std::string directoryName = "sample-directory";
+  std::string fileName = "sample-file";
+
+  // Initializing a ServiceClient that can then initialize the FileSystemClient or list file
+  // systems.
+  auto serviceClient = ServiceClient::CreateFromConnectionString(GetConnectionString());
+
+  // List all file systems.
+  std::string continuation;
+  std::vector<FileSystem> fileSystems;
+  do
+  {
+    auto response = serviceClient.ListFileSystems();
+    if (response->Continuation.HasValue())
+    {
+      continuation = response->Continuation.GetValue();
+    }
+    fileSystems.insert(
+        fileSystems.end(), response->Filesystems.begin(), response->Filesystems.end());
+  } while (!continuation.empty());
+
+  // Create file systems.
+  auto fileSystemClient = serviceClient.GetFileSystemClient(fileSystemName);
+  auto iter = std::find_if(
+      fileSystems.begin(), fileSystems.end(), [&fileSystemName](const FileSystem& fileSystem) {
+        return fileSystem.Name == fileSystemName;
+      });
+  if (iter == fileSystems.end())
+  {
+    fileSystemClient.Create();
+  }
```

Comment:
I feel this is too complicated as a sample. Sample is supposed to be easy and succinct, so that users can grasp the main points in a short time.

Creating a new filesystem and finding duplicate, which is the one we want to tell a beginner first?